### ISSUE TITLE
Add canaddconstraint

### ISSUE THREAD

**Issue by joaquimg** [JuliaOpt/MathOptInterface.jl#123]
see https://github.com/JuliaOpt/MathOptInterface.jl/pull/122#issuecomment-322765297

**Comment by blegat** [comment]
We should probably also add
```julia
canaddconstraint(m, ::Type{<:AbstractFunction}, ::Type{<:AbstractSet})
```
One example of usage would be the fallback of `cantransformconstraint` which would only know the type of the function.

**Comment by joaquimg** [comment]
closed by #129
